app/models/state_changer/stock_tube.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module StateChanger
+  # Handles the basic transitions of stock tubes within the library prep process
+  class StockTube < StateChanger::TubeBase
+    TERMINATED_STATES = %w[cancelled failed].freeze
+
+    private
+
+    def transfer_requests
+      @transfer_requests ||= labware.transfer_requests_as_target
+                                    .where.not(state: TERMINATED_STATES)
+    end
+
+    def update_associated_requests
+      return unless target_state == 'failed'
```

Comment:
on the plate equivalent I think you've got `if target_state == cancelled` ? is there a reason for the reversal of the logic?

Reply:
Plate seems to be `if target_state == 'failed'` but gates calling of `update_associated_requests` rather than acting as a guard clause. But `InitialStockTube` does use  return if target_state == 'cancelled'.

I need to have a quick step through the logic (as they may behave differently for states like 'started') but IIRC I should be able to standardize.

Reply:
Yep, its partly to handle the transition to started in those cases. (This is one of those cases where I *think* we can simplify, as the transfer requests should handle that themselves)

Reply:
Hmm, okay, realising I need to drastically extend the test coverage here. I was mostly relying on the existing tests as I refactored, but could do with some explicit ones to ensure coverage

Reply:
Auto-generated some tests to cover every permutation of the existing behaviour. Will use these to refactor, and then look at introducing something a little more sane.

Reply:
Auto tests there, and refactor done. But in doing so I noticed that `StateChanger::InitialStockTube` had an untriggered bug (It would do the wrong thing in the event of a state change we don't actually use on tubes).